src/pages/Admin/Venue/DetailsForm.tsx

```diff
@@ -805,8 +841,67 @@ const DetailsFormLeft: React.FC<DetailsFormLeftProps> = ({
     </div>
   );
 
+  const { sovereignVenueId } = useSovereignVenueId({ venueId });
+
+  const { currentVenue: venue } = useConnectCurrentVenueNG(sovereignVenueId);
+
+  const [userStatuses, setUserStatuses] = useState<UserStatus[]>([]);
+
+  const {
+    isShown: hasUserStatuses,
+    hide: hideUserStatuses,
+    show: showUserStatuses,
+  } = useShowHide(venue?.showUserStatus);
+
+  useEffect(() => {
+    if (!venue) return;
+    const venueUserStatuses = venue?.userStatuses ?? [];
+
+    setUserStatuses(venueUserStatuses);
+
+    if (venue.showUserStatus) {
+      showUserStatuses();
+    }
+  }, [showUserStatuses, venue]);
+
+  const removeUserStatus = (index: number) => {
+    const statuses = [...userStatuses];
+    statuses.splice(index, 1);
+    setUserStatuses(statuses);
+  };
+
+  const toggleUserStatus = useCallback(() => {
+    hasUserStatuses ? hideUserStatuses() : showUserStatuses();
+  }, [hasUserStatuses, hideUserStatuses, showUserStatuses]);
+
+  const addUserStatus = () =>
+    setUserStatuses([...userStatuses, { status: "", color: "#4BCC4B" }]);
+
+  const updateStatusColor = (color: string, index: number) => {
+    const statuses = [...userStatuses];
+    statuses[index] = { status: statuses[index].status, color: color };
+    setUserStatuses(statuses);
+  };
+
+  const updateStatusText = (
+    event: React.FormEvent<HTMLInputElement>,
+    index: number
+  ) => {
+    const statuses = [...userStatuses];
+    statuses[index] = {
+      color: statuses[index].color,
+      status: event.currentTarget.value,
```

Comment:
🤔 question to consider. Should we care about the unique statuses and apply the validation? Or we're okay to rely on users? (personally, would prefer validation)

Reply:
I'm not 100% sure here. Theoretically these can be anything, but maybe we should restrict them to use a-z0-9 only

Reply:
I meant the validation of the status uniqueness. Not to have 2 "online" statuses, for example.

Reply:
I guess we can add some basic validation that compares if the value already exists. But they can always work around it and add `.` or something else to trick the validation. 

Ex:
`online`
`online.`
`is online`
...etc.

Reply:
:) I think I meant a bit different side of things.
If we have 2 "online" statuses it means that when we choose 2nd of them in the dropdown it won't be set. Instead, we'll have the 1st one set. As we don't have IDs of the statuses we can't distinguish them properly. Wouldn't it be confusing?
https://github.com/sparkletown/sparkle/blob/7bb0d8cce2730d6ad9b651264c5b7ffb0650b0d0/src/hooks/useVenueUserStatuses.ts#L16:L18

```ts
  const userStatus = statuses.find(
    (userStatus) => userStatus.status === user?.status
  );
```
what do you think about it?

Reply:
It makes perfect sense, I will add some kind of basic validation when I get to it 💯

Reply:
I added some basic validation in: https://github.com/sparkletown/sparkle/commit/1945e0ca5749e776d75bc25b4cf621e7a8fdf8c2

I also added `@debt` because all of this should be refactored when moved to admin v3.